src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -12,13 +69,490 @@ import { ActivatedRoute } from '@angular/router';
+  }
 
-  constructor(private route: ActivatedRoute) { }
+  get allPresentSectionDetails(): SectionDetails[] {
+    let allPresentSectionDetails: SectionDetails[] = [];
+
+    this.allCheckedCourseDetails.forEach((courseDetails: CourseDetails) => {
+      allPresentSectionDetails = allPresentSectionDetails.concat(courseDetails.sections);
+    });
+
+    return allPresentSectionDetails;
+  }
+
+  get allCheckedSectionDetails(): SectionDetails[] {
+    return this.allPresentSectionDetails.filter((sectionDetails: SectionDetails) => sectionDetails.isChecked);
+  }
+
+  get allPresentTeamDetails(): TeamDetails[] {
+    let allPresentTeamDetails: TeamDetails[] = [];
+
+    this.allCheckedSectionDetails.forEach((sectionDetails: SectionDetails) => {
+      allPresentTeamDetails = allPresentTeamDetails.concat(sectionDetails.teams);
+    });
+
+    return allPresentTeamDetails;
+  }
+
+  get allCheckedTeamDetails(): TeamDetails[] {
+    return this.allPresentTeamDetails.filter((teamDetails: TeamDetails) => teamDetails.isChecked);
+  }
+
+  get allPresentStudentDetails(): StudentDetails[] {
+    let allPresentStudentDetails: StudentDetails[] = [];
+
+    this.allCheckedTeamDetails.forEach((teamDetails: TeamDetails) => {
+      allPresentStudentDetails = allPresentStudentDetails.concat(teamDetails.students);
+    });
+
+    return allPresentStudentDetails;
+  }
+
+  get allCheckedStudentDetails(): StudentDetails[] {
+    return this.allPresentStudentDetails.filter((studentDetails: StudentDetails) => studentDetails.isChecked);
+  }
+
+  get isAllPresentCoursesChecked(): boolean {
+    if (this.allPresentCourses.length === 0) {
+      return false;
+    }
+    return this.allPresentCourses.length === this.allCheckedCourses.length;
+  }
+
+  get isAllPresentCourseDetailssChecked(): boolean {
+    if (this.allPresentCourseDetails.length === 0) {
+      return false;
+    }
+    return this.allPresentCourseDetails.length === this.allCheckedCourseDetails.length;
+  }
+
+  get isAllPresentSectionDetailssChecked(): boolean {
+    if (this.allPresentSectionDetails.length === 0) {
+      return false;
+    }
+    return this.allPresentSectionDetails.length === this.allCheckedSectionDetails.length;
+  }
+
+  get isAllPresentTeamDetailsChecked(): boolean {
+    if (this.allPresentTeamDetails.length === 0) {
+      return false;
+    }
+    return this.allPresentTeamDetails.length === this.allCheckedTeamDetails.length;
+  }
+
+  get isAllPresentStudentDetailsChecked(): boolean {
+    if (this.allPresentStudentDetails.length === 0) {
+      return false;
+    }
+    return this.allPresentStudentDetails.length === this.allCheckedStudentDetails.length;
+  }
+
+  get isAnyPresentCoursesChecked(): boolean {
+    if (this.allPresentCourses.length === 0) {
+      return false;
+    }
+    return this.allCheckedCourses.length > 0;
+  }
+
+  courseStudentListSectionDataMap: Map<string, StudentListSectionData[]> = new Map();
```

Comment:
Use `{ [key: string]: StudentListSectionData[] }` as the type instead; it does not need to handle `| undefined` type.

Reply:
I changed this based on the review, but i dont really get the part `it does not need to handle | undefined` type. If i use the object and I want to obtain a value given a key, I suppose I still need to check if this value exists in the object...? If this is the case, I prefer using Map, as it comes with a lot of other methods that make things more convenient.